File: state_machines/nim/coin_state_machine.nim
var state = 0

while state != 25:
  echo "Balance: " , state
  echo "Insert a coin (penny, nickel, dime, quarter)"

  let coin = readLine(stdin)

  case coin
    of "penny":
      state += 1
    of "nickel":
      state += 5
    of "dime":
      state += 10
    of "quarter":
      state += 25
    else:
      echo "Not a coin"

echo "Here's your quarter"

File: state_machines/rust/coin_state_machine.rs
use std::io;

fn main() {
    let mut state: i32 = 0;

    while state != 25 {
        let mut coin = String::new();

        println!("Balance: {}", state);
        println!("Insert a coin (pennt, nickel, dime, quarter)");
        
        io::stdin().read_line(&mut coin)
                .expect("Failed to read line");

        match coin.trim() {
            "penny" => state += 1,
            "nickel" => state += 5,
            "dime" => state += 10,
            "quarter" => state += 25,
            _ => println!("Not a coin"),
        };
    }

    println!("Here's your quarter");
}
